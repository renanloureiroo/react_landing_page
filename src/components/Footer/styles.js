import styled, { css } from 'styled-components';

import { TextComponent } from '../TextComponent/index';
export const Container = styled.div`
  ${({ theme }) => css`
    & a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.sizes.small};
    }
    text-align: center;
  `}
`;

export const TextFooter = styled(TextComponent).attrs('div')``;
