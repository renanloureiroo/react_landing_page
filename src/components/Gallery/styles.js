import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css``}
`;
export const TextContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacing.huge};
  `}
`;
export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    justify-content: center;
    gap: ${theme.spacing.medium};

    & div {
      width: 100%;
      height: 100%;
      margin: 0 auto;
      overflow: hidden;
    }

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;
export const Grid = styled.img`
  transition: all 300ms ease-in-out;
  ${() => css`
    width: 100%;
    height: 100%;

    &:hover {
      transform: scale(1.2) rotate(7deg);
    }
  `}
`;
