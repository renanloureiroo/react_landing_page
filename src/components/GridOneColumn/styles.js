import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    max-width: 50%;
    margin: 0 auto;
    text-align: center;
  `}
`;
