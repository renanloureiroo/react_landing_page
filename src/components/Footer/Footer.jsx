import P from 'prop-types';
import * as Styled from './styles';

import { SectionContainer } from '../SectionContainer';

export const Footer = ({ children }) => {
  return (
    <SectionContainer>
      <Styled.Container>
        <Styled.TextFooter>{children}</Styled.TextFooter>
      </Styled.Container>
    </SectionContainer>
  );
};

Footer.propTypes = {
  children: P.node.isRequired,
};
