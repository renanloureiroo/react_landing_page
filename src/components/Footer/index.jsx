import P from 'prop-types';
import * as Styled from './styles';

import { SectionContainer } from '../SectionContainer';

export const Footer = ({ html }) => {
  return (
    <SectionContainer>
      <Styled.Container>
        <Styled.TextFooter>{html}</Styled.TextFooter>
      </Styled.Container>
    </SectionContainer>
  );
};

Footer.propTypes = {
  html: P.string.isRequired,
};
