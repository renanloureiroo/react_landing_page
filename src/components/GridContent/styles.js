import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    margin: 0 auto;
    max-width: 58rem;
    grid-template-columns: 1fr;
    text-align: center;
    justify-content: center;
    ${Title} {
      margin-bottom: ${theme.spacing.large};
    }
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
  & p{
    margin-bottom: ${theme.spacing.large}};
  }
  `}
`;
