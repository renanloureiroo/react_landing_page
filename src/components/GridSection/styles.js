import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-rows: 1fr 2fr;
    gap: ${theme.spacing.large};

    ${Title} {
      margin-bottom: ${theme.spacing.small};
    }

    @media ${theme.media.lteMedium} {
      display: block;
    }
  `}
`;
export const TextContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacing.large};
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap: ${theme.spacing.medium};
    counter-reset: grid-counter;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;
export const Grid = styled.div`
  ${({ theme }) => css`
    max-width: 36rem;
    ${Title} {
      position: relative;
      left: 5rem;
    }

    ${Title}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.spacing.xhuge};
      font-weight: bold;
      transform: rotate(10deg);
      left: -5rem;
      top: -2rem;
    }
  `}
`;
