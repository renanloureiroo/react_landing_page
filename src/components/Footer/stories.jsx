import { Footer } from './Footer';

export default {
  title: 'Footer',
  component: Footer,
  args: {
    children:
      '<p><a href="https://www.instagram.com/renanloureiroo">&nbsp;@renanloureiroo</a><br>&nbsp;</p>',
  },
};

export const Template = (args) => {
  return (
    <div>
      <Footer {...args} />
    </div>
  );
};
