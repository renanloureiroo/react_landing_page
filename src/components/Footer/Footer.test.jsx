import { renderTheme } from '../../styles/renderTheme';
import { Footer } from './Footer';

describe('<Footer />', () => {
  const html =
    '<p><a href="https://www.instagram.com/renanloureiroo">&nbsp;@renanloureiroo</a><br>&nbsp;</p>';
  it('should render', () => {
    const { container } = renderTheme(<Footer html={html} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
