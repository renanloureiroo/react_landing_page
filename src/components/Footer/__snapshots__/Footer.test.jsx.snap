// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render 1`] = `
.c2 {
  font-size: 1.6rem;
  line-height: 2.4rem;
}

.c1 {
  text-align: center;
}

.c1 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
}

.c0 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  padding: 3.2rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <p
      class="c2"
    >
      <p>
        <a
          href="https://www.instagram.com/renanloureiroo"
        >
           @renanloureiroo
        </a>
        <br />
         
      </p>
    </p>
  </div>
</div>
`;
